wordpress-mobile/WordPress-Android
# [Nullability Annotations to Java Classes] Replace `findViewById` with `ViewBinding` for History (`relatively safe`)

Parent #18906

This PR's main focus is to replace `findViewById ` with `ViewBinding` for the [HistoryDetailContainerFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/5b0d3f39c8cfe9c08b4e1547e46bc947f9be6b8f/WordPress/src/main/java/org/wordpress/android/ui/history/HistoryDetailContainerFragment.java#L61) class.

Additionally, this PR is also dealing with adding any missing nullability annotations to this fragment class, focusing mainly on its fields, and, any effected neighbour classes. But first, all existing warnings on this class are being resolved/suppressed so as to make it easier to deal with any missing nullability checks (with help from the IDE), especially when a `@Nullable` annotation is added.

FYI: This change is `relatively safe`, meaning that although there are compile-time changes associated with this change, and it needs testing, the changes included in this PR are very targeted and specific to one screen, and one screen only. Thus, if this screen is tested appropriately, it should be safe to merge this to `trunk`.

-----

PS: @RenanLukas I added you as the main reviewer, randomly so, since I just wanted someone from the Jetpack/WordPress mobile team to be aware of and sign-off on that change for JP/WPAndroid. Feel free to merge this PR directly yourself if you deem so.

-----

`findViewById ` -> `ViewBinding`

1. [Replace findviewbyid with viewbinding for history detail](https://github.com/wordpress-mobile/WordPress-Android/pull/19172/commits/85f6586cd0d079ec28b4ab0564fbada8f9d4affd)
2. [Use non-null viewbinding method parameter for history detail](https://github.com/wordpress-mobile/WordPress-Android/pull/19172/commits/6512ed28f0ab2466de6f3a824258198a98fa680a)

Nullability Annotation List:

1. [Add missing nullability annotations to sdk override methods](https://github.com/wordpress-mobile/WordPress-Android/pull/19172/commits/7c204435590d6aa2250811383f37337d0ba46209)
2. [Remove adapter from field and use it as non-null](https://github.com/wordpress-mobile/WordPress-Android/pull/19172/commits/e28be1d240e52e437756938bbcbcb7709e198c62)
3. [Remove revisions from field and use it as nullable](https://github.com/wordpress-mobile/WordPress-Android/pull/19172/commits/01bb24e9d2976ce67242b397cffaf1c1ec9f2ebc)
4. [Add missing nl-a to on page change listener field](https://github.com/wordpress-mobile/WordPress-Android/pull/19172/commits/e8112617b0b752d423f9e9e5c2ad0f5ecdf10d26)
5. [Add missing nl-a to on revision field](https://github.com/wordpress-mobile/WordPress-Android/pull/19172/commits/102dc63f4b2cec66e7470ee2f5643e8af3e57a78)

Warnings Resolution List:

1. [Make inner history detail fragment adapter class static](https://github.com/wordpress-mobile/WordPress-Android/pull/19172/commits/16d535f411db63703cf30a0a9d4b1b666f8df353)
2. [Replace anonymous class with lambda for history detail](https://github.com/wordpress-mobile/WordPress-Android/pull/19172/commits/5e643b5347cd6f65a98ee3da3515837a8197a5f5)

Warnings Suppression List:

1. [Suppress deprecation warnings on history detail](https://github.com/wordpress-mobile/WordPress-Android/pull/19172/commits/c508f268f75be002b5a772f55088f2a2dac80aaf)

-----

## To test:

1. Quickly smoke test any history related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected.
8. In addition to the above smoke test, you can expand the below and follow the inner and more explicit test steps within:

<details>
    <summary>History Details Screen [HistoryDetailContainerFragment.java]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` app.

- Go to `Posts` screen and tap on any post.
- Find the options menu (`top-right`) and tap on `History` .
- From the `History` screen, tap on any revision.
- Verify that the `History Details` screen is shown and functioning as expected.

</details>

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

6. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

9. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR submission checklist:


-----

## UI Changes testing checklist:
